service/service.go

```diff
@@ -43,8 +44,27 @@ func (svc *Service) Init(ctx context.Context, cfg *config.Config, buildTime, git
 	}
 
 	svc.Cfg = cfg
-	svc.ZebedeeCli = GetZebedee(cfg)
-	svc.DatasetCli = GetDatasetClient(cfg)
+
+	if svc.Producer, err = GetKafkaProducer(ctx, cfg.Kafka); err != nil {
+		return fmt.Errorf("failed to create kafka producer: %w", err)
+	}
+
+	// Initialize Zebedee client only if enabled
+	if cfg.EnableZebedeeCallbacks {
+		svc.ZebedeeCli = GetZebedee(cfg)
+	} else {
+		svc.ZebedeeCli = nil
+		log.Info(ctx, "Zebedee callbacks are disabled")
```

Comment:
events should be lowercase also should say what is actually happening here, not just why (same for datasets logging below)
```suggestion
		log.Info(ctx, "healthchecks skipped for zebedee as callbacks disabled ")
```
https://github.com/ONSdigital/dp-standards/blob/main/LOGGING_STANDARDS.md#effective-event-types